### PR TITLE
HTTPCLIENT-2277: removal of deprecated cache functionality / logic that does not belong to the caching layer

### PR DESCRIPTION
**Pull request by ok2c**
This change-set includes
* removal of deprecated cache functionality
* removal of HTTP protocol logic that does not belong to the caching layer

### CODE REVIEW

File: httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/ResponseProtocolCompliance.java [hunk 1]
```diff
@@ -28,198 +28,51 @@
...
+
+/**
+ * This response interceptor removes common {@literal Content-} headers from the 304
+ * response messages based on the provision of the HTTP specification that 304
+ * status code represents a confirmation that the requested resource content has
+ * not changed since its retrieval or the last update and adds {@literal Date}
+ * header if it is not present in the response message..
+ */
+@Contract(threading = ThreadingBehavior.IMMUTABLE)
+class ResponseProtocolCompliance implements HttpResponseInterceptor {
```
Comment (arturobernalg) [inline httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/ResponseProtocolCompliance.java:51]: Is this change related to the guidelines in '[3.2. Updating Stored Header Fields'](https://www.rfc-editor.org/rfc/rfc9111.html#name-updating-stored-header-fiel) from the RFC9111 specification?

File: httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/CachingExecBase.java [hunk 2]
```diff
@@ -187,8 +164,7 @@ SimpleHttpResponse generateCachedResponse(
             final ResponseCacheControl responseCacheControl,
```
Comment (arturobernalg) [inline httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/CachingExecBase.java:164]: I think we can remove `responseCacheControl`

**Review by arturobernalg — CHANGES_REQUESTED** [review]
@ok2c, it looks good. Just a few minor comments.

**Comment by ok2c** [comment]
@arturobernalg All sounds reasonable. Please do another pass.

**Review by arturobernalg — APPROVED** [review]
The change-set looks good

**Pull request merged**